File: backend/models/MedicalRecord.js
const mongoose = require('mongoose');

const MedicalRecordSchema = new mongoose.Schema({
  patientId: { type: String, required: true },
  patientName: { type: String, required: true },
  doctorId: { type: String, required: true },
  doctorName: { type: String, required: true },
  appointmentId: { type: String, required: true },
  diagnosis: { type: String, required: true },
  prescription: { type: String, required: true },
  notes: { type: String, default: "" },
  followUpDate: { type: String, default: "" },
  createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('MedicalRecord', MedicalRecordSchema);

File: app/src/main/java/com/example/medplus/model/MedicalRecord.kt
package com.example.medplus.model

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class MedicalRecord(
    @SerializedName("_id") val recordId: String = "",
    val patientId: String = "",
    val patientName: String = "",
    val doctorId: String = "",
    val doctorName: String = "",
    val appointmentId: String = "",
    val diagnosis: String = "",
    val prescription: String = "",
    val notes: String = "",
    val followUpDate: String = "",
    val createdAt: String = ""
) : Serializable
